postreise/plot/plot_carbon_bar.py

```diff
@@ -1,51 +1,77 @@
 import numpy as np
 from matplotlib import pyplot as plt
+from powersimdata.scenario.scenario import Scenario
 
+from postreise.analyze.generation.emissions import (
+    generate_emissions_stats,
+    summarize_emissions_by_bus,
+)
 
-def plot_carbon_bar(carbon_by_bus_1, carbon_by_bus_2, scenario_name_1, scenario_name_2):
-    """Makes bar chart of carbon emissions by fuel type for two scenarios.
 
-    :param dict carbon_by_bus_1: emission by type and bus for first scenario
-        as returned by :func:`postreise.analyze.carbon.summarize_carbon_by_bus`.
-    :param dict carbon_by_bus_2: emission by type and bus, for second scenario
-        as returned by :func:`postreise.analyze.carbon.summarize_carbon_by_bus`.
-    :param str scenario_name_1: name of first scenario, for labeling plot.
-    :param str scenario_name_2: name of second scenario, for labeling plot.
+def plot_carbon_bar(*args, labels=None, labels_size=15):
+    """Make bar chart of carbon emissions by fuel type for n scenarios.
+
+    :param powersimdata.scenario.scenario.Scenario args: scenario instances.
+    :param list/tuple/set labels: labels on plot. Default to scenario id.
+    :param int labels_size: size of labels.
+    :raises ValueError:
+        if ``args`` are not scenario instances.
+        if ``labels`` has a different length than the number of passed scenarios.
+    :raises TypeError:
+        if ``labels`` is not an iterable.
+        if ``labels_size`` is not an int.
```

Comment:
Do we want to follow the convention: always return the `axes` of the plot and provide option to not show the plot immediately after the function call? Same comment for all the plot functions being refactored here.

Reply:
The plotting functions that are in the tutorial and that don't have the `axes` returned and the `show_plot` parameters are:
* `plot_bar_generation_vs_capacity`
* `plot_bar_shortfall`
* `map_carbon_emission_generator` and `map_carbon_emission_difference`
* `plot_curtailment_time_series`
* `plot_n_scenarios`
* `plot_generation_time_series_stack`
* `plot_heatmap`
* `map_interconnections`
* `map_lmp`
* `plot_pie_generation_vs_capacity`
* `plot_powerflow_snapshot`
Should I go ahead and refactor them?